enhancements/kubernetes-nmstate-integration.md

```diff
@@ -0,0 +1,165 @@
+Then, the namespace, RBAC, and finally the DaemonSet are applied. Custom resources of kind `NMstate`
+with other names will be ignored.
+
+A new kubernetes-nmstate handler DaemonSet is deployed in the cluster by the operator.
+This DaemonSet contains nmstate package and interacts with the NetworkManager
+on the host by mounting the related dbus. The project contains three
+Custom Resource Definitions, `NodeNetworkState`, `NodeNetworkConfigurationPolicy` and
+`NodeNetworkConfigurationEnactment`. `NodeNetworkState` objects are created per each
+node in the cluster and can be used to report available interfaces and network configuration.
+These objects are created by kubernetes-nmstate and must not be touched by a user.
+`NodeNetworkConfigurationPolicy` objects can be used to specify desired
+networking state per node or set of nodes. It uses API similar to `NodeNetworkState`. 
+`NodeNetworkConfigurationEnactment` objects will be created per each node, per each matching
+`NodeNetworkConfigurationPolicy`.
+
+kubernetes-nmstate DaemonSet creates a custom resource of `NodeNetworkState` type per each node and
+updates the network topology from each OpenShift node.
+
+User configures host network changes and apply a policy in `NodeNetworkConfigurationPolicy` custom
+resource. Network topology is configured via `desiredState` section in `NodeNetworkConfigurationPolicy`.
+Multiple `NodeNetworkConfigurationPolicy` custom resources can be created.
+
+Upon receiving a notification event of `NodeNetworkState` update,
+kubernetes-nmstate Daemon verify the correctness of `NodeNetworkState` custom resource and
+apply the selected profile to the specific node.
+
+`NodeNetworkConfigurationEnactment` objects is a read-only object that represents Policy exectuion
+per each matching Node. It will expose configuration status per each Node.
+
+A new container image (kubernetes-nmstate-handler) will be created.
+
+A new container image (kubernetes-nmstate-operator) will be created. However, the operator
+and the handler will co-exist in the same upstream repo.
+
+An upstream API group of nmstate.io is currently used.
+
+### User Stories
+
+#### Bond creation
+
+* Be able to create bond interfaces on OpenShift nodes.
+ 
+  Typically, MCO + Ignition would be utilized. However, in a Day 1 install, that may not be possible.
+  It may be necessary for the bond/vlan configuration be done before Ignition can even be reached.
+  
+* Create a vlan interface on top of the bond inter.
+
+#### Assign ip address
+
+* Assign static and/or dynamic ip address on interfaces
+* Assign ipv4 and/or ipv6
+
+#### Create/Update/Remove network routes
+
+* Be able to Create/Update/Remove network routes for different interfaces like (bond,ethernet,sriov vf and sriov pf)
+
+#### Manage/Configure host SR-IOV network device
+
+* Be able to change host Virtual functions configuration (not managed by the sriov-operator) like vlans mtu driver etc..
```

Comment:
Why would I want to do this?

Reply:
Currently the reason is a bit vague because there is a CNI plugin in the picture.
Being able to control SRIOV in general is partially a node level resource, so it makes sense to provide the means for controlling HW at this level. One application could be to run other services on the node (daemonsets?) that need special network access or in case one wants to use VF/s as regular secondary interfaces. Another option would be to control a NIC level setting that the CNI does not.

Maybe there are more interesting scenarios.